### PR TITLE
added service for custom donation list by campaign

### PR DESCRIPTION
**Pull request by nradkova**
`donation/recent/by-campaign/:id`
-- new endpoint and service added for query filtering donations by `ampaignId,` status - `succeeded`, sorted desc by `createdAt`.  
-- used to fix issue `#566 in frontend`

### CODE REVIEW

File: apps/api/src/donations/donations.service.ts [hunk 1]
```diff
@@ -167,4 +167,19 @@ export class DonationsService {
     }, 0)
     return { donations, total }
   }
+
+  async getRecentDonationsByCampaign(campaignId: string) {
+    const vaults = await this.prisma.vault.findMany({ where: { campaignId } })
+    const donations = await this.prisma.donation.findMany({
+      where: {
```
Comment (Bunny303) [inline apps/api/src/donations/donations.service.ts:174]: Request to db for getting vaults could be skipped with joining vault table. Where clause could be modified 
where: { targetVault: { campaignId },  status: DonationStatus.succeeded }

File: apps/api/src/donations/donations.controller.ts [hunk 2]
```diff
@@ -102,4 +102,10 @@ export class DonationsController {
   ) {
     return this.donationsService.softDelete(idsToDelete)
   }
+
+  @Get('recent/by-campaign/:id')
+  @Public()
+  async getLastTenStripeDonationsByCampaign(@Param('id') id: string) {
```
Comment (Bunny303) [inline apps/api/src/donations/donations.controller.ts:108]: Name could be changed. There is no limimt set in the query for donations so it won't return only 10 if more than that exists.

File: apps/api/src/donations/donations.controller.ts [hunk 3]
```diff
@@ -102,4 +102,10 @@ export class DonationsController {
   ) {
     return this.donationsService.softDelete(idsToDelete)
   }
+
+  @Get('recent/by-campaign/:id')
```
Comment (Bunny303) [inline apps/api/src/donations/donations.controller.ts:106]: Route could be 'list/:campaignId' or 'by-campaign/:campaignId'.

File: apps/api/src/donations/donations.service.ts (hunk 1 above)
Comment (nradkova) [reply apps/api/src/donations/donations.service.ts:174]: Thank you for the optimization.

File: apps/api/src/donations/donations.controller.ts (hunk 2 above)
Comment (nradkova) [reply apps/api/src/donations/donations.controller.ts:108]: This is really a typo as it was initially limited to 10 results. Thanks.

Comment (Bunny303) [reply apps/api/src/donations/donations.controller.ts:108]: Actually will be good to have limit set in the query based on the needs of the front end.

Comment (nradkova) [reply apps/api/src/donations/donations.controller.ts]: In the requirements is set to be shown all benefactors and donations amount -- `Виж всички`.  That's why the query is changed to take all of them. I left the limitation in the front end with a comment, in order to be discussed.

**Pull request closed without merging**